va/va_cp.h

```diff
@@ -0,0 +1,420 @@
+            /** \brief Reserved for future use, must be zero */
+            uint32_t reserved                             : 1;
+            } bits;
+        uint32_t value;
+    } header_fields; 
+     /** \brief Same as the VP9 bitstream syntax element. */
+    uint16_t frame_width_minus1;
+     /** \brief Same as the VP9 bitstream syntax element. */
+    uint16_t frame_height_minus1;
+    /** \brief Same as the VP9 bitstream syntax element. */
+    uint16_t render_width_minus1;
+    /** \brief Same as the VP9 bitstream syntax element. */
+    uint16_t render_height_minus1;
+    /** \brief Same as the VP9 bitstream syntax element. */
+    uint8_t refresh_frame_flags;
+    /** \brief Reserved bytes for future use, must be zero */
+    uint32_t va_reserved[VA_PADDING_MEDIUM];
+} VACencSliceParameterBufferVP9;
+
+/** \brief Cenc Slice Buffer Type*/
+typedef enum {
+    /** \brief Parsed slice parameters \c VACencSliceParameterBuffer* */
+    VaCencSliceBufParamter = 1,
+    /** \brief Raw slice header of bitstream*/
+    VaCencSliceBufRaw      = 2
+}VACencSliceBufType;
+
+/** \brief Buffer for CENC status reporting*/
+typedef struct _VACencStatusBuf
+{
+    /** \brief Encryption status. VA_ENCRYPTION_STATUS_SUCCESSFUL if 
+     *  hardware has returned detailed inforamtion, others mean the 
+     *  CENC result is invalid  */
+    VAEncryptionStatus status;
+    /* \brief status report feed back number */
+    uint16_t           status_report_feedback_number;
+    /** \brief Buf size in bytes.  0 means buf is invalid*/
+    uint32_t           buf_size;
+    /** \brief Buffer formatted as raw data from bitstream for sequence parameter,
+     *  picture parameter, SEI parameters. Or \c NULL means buf is invalid.*/
+    void               *buf;
+    /** \brief Slice buffer type, see*/
+    VACencSliceBufType slice_buf_type;
+    /** \brief Slice buffer size in bytes. 0 means slice_buf is invalid*/
+    uint32_t           slice_buf_size;
+    /** \brief Slice buffer, parsed slice header information. Or \c NULL
+     *  means slice_buf is invalid.*/
+    void               *slice_buf;
+    /** \brief Reserved bytes for future use, must be zero */
+    unsigned long va_reserved[VA_PADDING_MEDIUM];
+} VACencStatusBuf;
+
+/**
+ * Block call to get cenc status report. 
+ * After the application sends out cenc workload, it must call vaQueryCenc
+ * to get cenc status report for decrypt result.
+ * upon the return, info will point to an array of \ref VACencStatusBuf structure,
+ * The array is terminated if "status==-1" is detected.
+ */
+VAStatus vaQueryCenc(
+    VADisplay dpy,
+    VABufferID buffer,
+    VACencStatusBuf *info
+);
+
+/**
+ * Make the end of rendering for Cenc decode for the encrypted bitstream. 
+ * The server should start all pending rendering operations and decryption 
+ * processing related with  this encrypted bitstream.  */
+VAStatus vaEndCenc (
+    VADisplay dpy,
+    VAContextID context
+);
```

Comment:
Does this need to be a separate call?  Since you're using the `vaRender*()` calls for the rest of it, it looks like `vaEndPicture()` should be sufficient (though I may well be missing something).